nucliadb/nucliadb/purge/__init__.py

```diff
@@ -142,6 +144,52 @@ async def purge_kb_storage(driver: Driver, storage: Storage):
     logger.info("FINISH PURGING KB STORAGE")
 
 
+async def purge_kb_vectorsets(driver: Driver, storage: Storage):
+    """Vectors for a vectorset are stored in a key inside each resource. Iterate
+    through all resources of the KB and remove any storage object containing
+    vectors for the specific vectorset to purge.
+
+    """
+    logger.info("START PURGING KB VECTORSETS")
+
+    purged = []
+    async for key in _iter_keys(driver, KB_VECTORSET_TO_DELETE_BASE):
+        logger.info(f"Purging vectorsets {key}")
+        try:
+            _base, kbid, vectorset = key.lstrip("/").split("/")
+        except ValueError:
+            logger.info(
+                f"  X Skipping purge {key}, wrong key format, expected {KB_VECTORSET_TO_DELETE}"
+            )
+            continue
+
+        try:
+            async with driver.transaction(read_only=True) as txn:
+                kb = KnowledgeBox(txn, storage, kbid)
+                async for resource in kb.iterate_resources():
+                    fields = await resource.get_fields(force=True)
+            # we don't need the maindb transaction anymore to remove vectors from storage
```

Comment:
nit: is this comment needed? 
As far as I see, at least you need the txn to iterate the resources no?

Reply:
I think it doesn't harm, is kind of a hacky optimization. Knowing we don't need the transaction anymore is not obvious. Anyway, I don't mind removing it